cartoframes/dataset.py

```diff
@@ -376,14 +382,85 @@ def get_columns(context, query):
 
 def get_query(dataset):
     if isinstance(dataset, Dataset):
-        return dataset.query or _default_query(dataset)
+        if dataset.query is None:
+            return _compute_query(dataset)
+        else:
+            return dataset.query
 
 
-def _default_query(dataset):
+def _compute_query(dataset):
     if dataset.table_name and dataset.schema:
         return 'SELECT * FROM "{0}"."{1}"'.format(dataset.schema, dataset.table_name)
 
 
+def get_geodataframe(dataset):
+    if isinstance(dataset, Dataset):
+        if dataset.gdf is None:
+            return _compute_geodataframe(dataset)
+        else:
+            return dataset.gdf
+
+
+def _compute_geodataframe(dataset):
+    if dataset.df is not None:
+        df = dataset.df.copy()
+        geom_column = _get_geom_column(df)
+        if geom_column is not None:
+            df['geometry'] = _compute_geometry_from_geom(geom_column)
+        else:
+            lat_column = _get_lat_column(df)
+            lng_column = _get_lng_column(df)
+            if lat_column is not None and lng_column is not None:
+                df['geometry'] = _compute_geometry_from_latlng(lat_column, lng_column)
+            else:
+                raise ValueError('DataFrame has no geographic data.')
```

Comment:
How about something like this:
"No geographic data found. If a geometry exists, change the column name to `the_geom` or `geom` or ensure it is a GeoDataFrame with a valid geometry. If there are latitude/longitude columns, rename to `lat`/`lng`."

This way people have a route forward if they do indeed have geographic information.

Reply:
Or better yet we can give them the full list of valid names/formats.

Reply:
What about 

> No geographic data found. If a geometry exists, change the column name (geometry, the_geom, wkt_geometry, wkb_geometry, geom, wkt, wkb) or ensure it is a DataFrame with a valid geometry. If there are latitude/longitude columns, rename to (latitude, lat), (longitude, lng, lon, long).

Reply:
👍